core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1271,7 +1271,7 @@ private[spark] object SparkSubmitUtils {
         // retrieve all resolved dependencies
         ivy.retrieve(rr.getModuleDescriptor.getModuleRevisionId,
           packagesDirectory.getAbsolutePath + File.separator +
-            "[organization]_[artifact]-[revision].[ext]",
+            "[organization]_[artifact]-[revision](-[classifier]).[ext]",
```

Comment:
In my example, 
```
zookeeper-jar: {artifact=zookeeper, ext=jar, module=zookeeper, classifier=tests, organisation=org.apache.zookeeper, type=test-jar, revision=3.4.6}
zookeeper-jar: {artifact=zookeeper, ext=jar, module=zookeeper, organisation=org.apache.zookeeper, type=jar, revision=3.4.6}
```

Both dependencies will have the same name `org.apache.zookeeper_zookeeper-3.4.6.jar` and cause the collision. After my PR, they will be different names
`org.apache.zookeeper_zookeeper-3.4.6.jar`
`org.apache.zookeeper_zookeeper-3.4.6-tests.jar`

Reply:
The reason why I am putting `classifier` at the end. I am just following the [default artifact partern](https://github.com/apache/ant-ivy/blob/12aeeec70feae05a87a5adfe7b9c2c63744be37f/src/java/org/apache/ivy/core/cache/DefaultRepositoryCacheManager.java#L83) in the Apache IVY.

Reply:
FWIW this looks fine; I needed a similar change as part of https://github.com/srowen/spark/commit/41267020701be4877be352e1678113bd9870ec12 
It's possible you may need some other changes from that WIP commit; not sure.

Reply:
Thanks!

Reply:
I tried it today. Somehow, I only got the test jar downloaded. Have you guys seen this issue?